### PR TITLE
Automatically update the library section after toggling the setting

### PR DESCRIPTION
**Pull request by oleitao**
For #3869

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@oleitao Wouldn't it be better to make the change occur without restarting the app?

File: Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs [hunk 1]
```diff
@@ -111,10 +112,48 @@ public bool ShowLibrarySection
...
                     App.AppSettings.ShowLibrarySection = value;
+                    if (showLibrarySection)
+                    {
+                        DiaplyContentDialog();
+                    }
                 }
             }
         }
 
+        private async void DiaplyContentDialog()
```
Comment (d2dyno1) [inline Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs:123]: `DisplayContentDialog`

File: Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs [hunk 2]
```diff
@@ -111,10 +112,48 @@ public bool ShowLibrarySection
...
+        {
+            ContentDialog dialog = new ContentDialog()
```
Comment (d2dyno1) [inline Files/ViewModels/SettingsViewModels/PreferencesViewModel.cs:125]: Could you use `DynamicDialog`.

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

File: Files/App.xaml.cs [hunk 3]
```diff
@@ -100,12 +100,16 @@ private static async Task EnsureSettingsAndConfigurationAreBootstrapped()
...
+                {
+                    await LibraryManager.EnumerateDrivesAsync();
+                }
+
                 await DrivesManager.EnumerateDrivesAsync();
                 await CloudDrivesManager.EnumerateDrivesAsync();
                 await NetworkDrivesManager.EnumerateDrivesAsync();
                 await WSLDistroManager.EnumerateDrivesAsync();
-            });
+            });
```
Comment (yaichenbaum) [inline Files/App.xaml.cs:112]:
```suggestion
            });
```

File: Files/Views/SettingsPages/Preferences.xaml.cs [hunk 4]
```diff
@@ -1,12 +1,29 @@
...
+        {
+            if (App.AppSettings.ShowLibrarySection)
+            {
+                await LibraryManager.EnumerateDrivesAsync();
+            }
+            else
+            {
+                await LibraryManager.RemoveEnumerateDrivesAsync();
+            }
+        }
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/Preferences.xaml.cs:27]: Can you move this code to the settings view model for preferences?

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/Views/SettingsPages/Preferences.xaml.cs (hunk 4 above)
Comment (oleitao) [reply Files/Views/SettingsPages/Preferences.xaml.cs:27]: yes, please check on my last comit

File: Files/Views/SettingsPages/Preferences.xaml.cs [hunk 5]
```diff
@@ -6,7 +6,13 @@ public sealed partial class Preferences : Page
...
         public Preferences()
         {
+
             InitializeComponent();
         }
+
+        private async void ShowLibrarySection_Toggled(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            App.AppSettings.ShowLibrarySection = ShowLibrarySection.IsOn;
+        }
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/Preferences.xaml.cs:16]:
```suggestion
```

File: Files/Views/SettingsPages/Preferences.xaml [hunk 6]
```diff
@@ -112,6 +112,7 @@
                     x:Name="ShowLibrarySection"
                     x:Uid="SettingsPreferencesShowLibrarySection"
                     Header="Show Library section on the sidebar"
+                    Toggled="ShowLibrarySection_Toggled"
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/Preferences.xaml:115]:
```suggestion
```

File: Files/Views/SettingsPages/Preferences.xaml.cs [hunk 7]
```diff
@@ -8,5 +8,6 @@ public Preferences()
         {
             InitializeComponent();
         }
+
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/Preferences.xaml.cs:11]:
```suggestion
```

**Comment by yaichenbaum** [comment]
@oleitao Can you make sure the library section shows up after the favorites section after turning it on?

**Comment by oleitao** [comment]
> @oleitao Can you make sure the library section shows up after the favorites section after turning it on?

Library section hide/show is not working because you removed event:
`private async void ShowLibrarySection_Toggled(object sender, Windows.UI.Xaml.RoutedEventArgs e) 
{
...
}`

**Comment by yaichenbaum** [comment]
@oleitao It's showing fine for me after toggling the setting, the only issue is that it shows at the bottom of the sidebar instead of after the favorites section.

File: Files/Views/SettingsPages/Preferences.xaml.cs (hunk 4 above)
Comment (yaichenbaum) [reply Files/Views/SettingsPages/Preferences.xaml.cs:14]: @gave92 Isn't this a little redundant since it's already bound to the property?

**Review by yaichenbaum — APPROVED** [review]
@oleitao Great work! LGTM

**Pull request merged**